network/src/connectivity_manager/mod.rs

```diff
@@ -316,8 +334,14 @@ where
             .filter(|peer_id| !eligible.contains_key(peer_id))
             .cloned()
             .collect();
-        for p in stale_dials.into_iter() {
-            self.dial_queue.remove(&p);
+
+        if !stale_dials.is_empty() {
+            send_struct_log!(StructuredLogEntry::new_named("cancel_stale_dials")
```

Comment:
same as above
but ... why are we adding this in this commit :O

Reply:
Oh, I'll cut it out :/. I spent a bunch of time chugging through adding this structured logging through my hunt after the seg fault that kept happening during the test.

Reply:
oh we have seg fault?

Reply:
I was getting seg faults during testing either by the structured logging, or by me seemingly dereferencing something in the logic through the channels.  Turning on structured logging made it worse :)

Reply:
rust claims it'd never seg fault without unsafe code